numba_dpex/core/runtime/_dpexrt_python.c

```diff
@@ -521,8 +549,9 @@ DPEXRT_MemInfo_alloc(npy_intp size, size_t usm_type, const char *device)
  */
 static NRT_MemInfo *DPEXRT_MemInfo_fill(NRT_MemInfo *mi,
                                         size_t itemsize,
-                                        bool is_float,
-                                        uint8_t value,
+                                        bool dest_is_float,
+                                        bool value_is_float,
+                                        int64_t value,
```

Comment:
Do we really need `int64_t`? Can you please evaluate with `uint64_t` and using the union for bit-pattern conversion?